### PR TITLE
Add another React HN implementation

### PR DESCRIPTION
**Pull request by stephenkingsley**
add a project
 - [react-hn-pwa](https://hn-pwa-d8b2e.firebaseapp.com/)
 - [lighthouse](https://stephenkingsley.github.io/hn-pwa-d8b2e.firebaseapp.com_2017-06-13_19-52-39.report.dom.html)

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
Hi Stephen,

Thank you for your submission. It's great to see new additions!

It looks like there are a few items you could add to get to parity with the baseline for [submissions](https://github.com/tastejs/hacker-news-pwas#specification). I've highlighted the items here from the list I didn't see.

**Views: Hacker News Top Stories, New, Show, Ask, Jobs & threaded Comments**
I see the top, show, and ask list views but don't see the new or jobs list views. Additionally, I don't see how to access threaded comments, or a "detail" view of a story.

**App must use the Application Shell pattern to instantly load the skeleton of the UI on repeat visits**
When loading from the Service Worker, I didn't see a skeleton of the UI. You might take a look at how this is implemented in a few of the other approaches for inspiration.

**Comment by housseindjirdeh** [comment]
Thanks for submitting this @stephenkingsley! As @kristoferbaxter mentioned there are a few submission requirements the apps need to meet - the other implementations are a great way to get an idea of how to do them. But please don't hesitate to ask if you need help in adding those features and we can point you in the right direction :)

**Comment by stephenkingsley** [comment]
@kristoferbaxter thx for your suggestion, I will make this submit better on the weekend. 

> App must use the Application Shell pattern to instantly load the skeleton of the UI on repeat visits
When loading from the Service Worker, I didn't see a skeleton of the UI. You might take a look at how this is implemented in a few of the other approaches for inspiration.

as your say this "Application Shell" is not enough, can you please tell me what's the good Application Shell? I will try my best to work on this submit.

ccc

**Comment by kristoferbaxter** [comment]
That looks like your application shell allright. Must have made a mistake when looking earlier!

**Comment by stephenkingsley** [comment]
@kristoferbaxter I add story page and Job，ask page in this application. please check in here [https://hn-pwa-d8b2e.firebaseapp.com/](https://hn-pwa-d8b2e.firebaseapp.com/)

File: site/_apps/react-hn-stephen.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+weight: 9
```
Comment (addyosmani) [inline site/_apps/react-hn-stephen.txt]: We will want to adjust the weights here to account for other apps that landed while this PR was being worked on.

Comment (stephenkingsley) [inline site/_apps/react-hn-stephen.txt]: ok, I modified the weight, and the value is 14

**Comment by stephenkingsley** [comment]
@addyosmani  review again please, I will try my best to work for this submit!

File: site/_apps/react-hn-stephen.txt (hunk 1 above)
Comment (ragingwind) [inline site/_apps/react-hn-stephen.txt]: @stephenkingsley I'm not pretty sure that the weight should be 15 in place of https://github.com/ragingwind/hacker-news-pwas/blob/3156b1edd2e33adbd254a455037af7a605fa6364/site/_apps/next-hnpwa.txt#L2

Comment (stephenkingsley) [inline site/_apps/react-hn-stephen.txt]: ok, I modified!

**Comment by stephenkingsley** [comment]
may be can review again?

**Comment by addyosmani** [comment]
@stephenkingsley Yep. I'll add this to my todo list for today.

**Comment by addyosmani** [comment]
Feedback:

Homepage/top-stories appears to be implemented as an infinite scrolling list instead of using pagination. For consistency with the other apps, could you consider switching to pagination like the following apps have?

https://react-hn.kristoferbaxter.com/
https://hn.kristoferbaxter.com/
https://hn.svelte.technology/

Formatting for each points/comments link appears to have some issues. Is it possible to format these to be a little more readable?

image

Super minor: 

There appear to be some stray console logs in the app
image

Cheers!

File: site/_apps/react-hn-stephen.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+other-details: 'create-react-app'
+authors:
+    - name: 'stephen'
+lighthouse: '82/100'
+interactive-em: '3.1s'
+interactive-faster-3g: '2.41s'
+lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=170613_VA_91fc5fd2466f353c1aec50e89c4ecffb&run=1
+wpt-em-link: https://www.webpagetest.org/result/170613_39_cf8dc6cebd9df75ccb2aef140e4e80ae/
+wpt-faster-3g-link: https://www.webpagetest.org/result/170613_VA_91fc5fd2466f353c1aec50e89c4ecffb/
+image: /assets/images/react-hn-stephen.png
```
Comment (addyosmani) [inline site/_apps/react-hn-stephen.txt]: The performance traces check out. Nice work!

**Comment by stephenkingsley** [comment]
@addyosmani thx you advices! and I fix it, now you can see [https://hn-pwa-d8b2e.firebaseapp.com/1](https://hn-pwa-d8b2e.firebaseapp.com/1)

please review again!

**Comment by stephenkingsley** [comment]
@addyosmani please review

**Comment by anubhav7495** [comment]
@stephenkingsley Hi. There are some things I would still consider missing from the app.

Please try to fix issues marked with either **High** or **Medium** priority before submission. **Low** priority issues can be worked out later.

#### Issues
1. Clicking on `Ask HN` links open a new tab with `NEWS` view. **[High]**
2. Posts are always numbered from `1` on subsequent pages in a view. `Page 2` posts should start from `31` and so on. **[High]**
3. Refreshing in app always reroutes to `NEWS` view irrespective of previously active view. **[High]**
4. Post title should wrap in mobile view but currently, overflows. Screenshot attached below. **[Medium]**
react-hn-1

5. Threaded comments should have usernames as well, looks odd otherwise. **[Medium]**
6. Stray `console.logs` still present. **[Medium]**
7. Please try to conditionally render Post metadata. Screenshot attached below. **[Medium]**
conditional-render

8. Header gives no clue as to which view is currently active, eg: `NEWS` or `NEW`. **[Low]**
9. 100% width on Desktop, getting cursor all the way to the left takes a little extra work. **[Low]**
10. The loading animation on the desktop view is way out of the viewport. **[Low]**

Thanks.

**Comment by addyosmani** [comment]
Side: @anubhav7495 Interested in becoming a collaborator/member of this repo? I've noticed some of your reviews have been pretty excellent recently 🥇

**Comment by anubhav7495** [comment]
@addyosmani Yes:heart_eyes:, and Thanks:grin:

**Comment by addyosmani** [comment]
@anubhav7495 done!

**Comment by addyosmani** [comment]
@stephenkingsley Do you think you might have time to go through some of the Medium/High items in our last review?

**Comment by stephenkingsley** [comment]
@addyosmani yeah, I will modify it, and submit another review in tomorrow

**Comment by stephenkingsley** [comment]
@addyosmani I fix the `High` and `Medium` issum. And the screen shot is in the bottom.

- [x] Clicking on Ask HN links open a new tab with NEWS view. [High]

- [x] Posts are always numbered from 1 on subsequent pages in a view. Page 2 posts should start from 31 and so on. [High]

- [x] Refreshing in app always reroutes to NEWS view irrespective of previously active view. [High]

- [x] Post title should wrap in mobile view but currently, overflows. Screenshot attached below. 

- [x] Threaded comments should have usernames as well, looks odd otherwise. [Medium]

- [x] Stray console.logs still present. [Medium] 
but I will logs something about service worker.
```js
console.log('New content is available; please refresh.');
console.log('Content is cached for offline use.');
```

- [x] Please try to conditionally render Post metadata. Screenshot attached below. [Medium]



fireshot capture 36 - hnpwa - http___127 0 0 1_3000_item_14825473

**Comment by addyosmani** [comment]
@stephenkingsley These looks solid - I'll take another look through this afternoon. We might want to also redo the Lighthouse + WebPageTest numbers to use the latest deploys (I'd also update weight to 20 if you can).

I'll hand over to @anubhav7495 for his take on the latest version.

**Comment by stephenkingsley** [comment]
@addyosmani the last version can merge?

**Comment by anubhav7495** [comment]
@stephenkingsley Thanks for working on the issues.

Some things I still feel can be improved:

1. `Ask HN` links should open the in-app comments view instead of Hacker News website.
2. Post meta data in `Jobs` view still looks the same way as before. Please try to conditionally render it.
3. Scroll position should reset to `0` on changing through pages.
4. Header link should indicate the currently active view since on mobile PWA experience address bar is not visible.

Check out these implementations for some help: [kristoferbaxter/react-hn](https://react-hn.kristoferbaxter.com/), [insin/react-hn](https://react-hn.appspot.com)

Thanks.

**Comment by stephenkingsley** [comment]
@anubhav7495 ok, I will fix those issue before sunday.

**Comment by stephenkingsley** [comment]
@anubhav7495 I fix those issue
- [x] Ask HN links should open the in-app comments view instead of Hacker News website.
- [x] Post meta data in Jobs view still looks the same way as before. Please try to conditionally render it.
- [x] Scroll position should reset to 0 on changing through pages.
- [x] Header link should indicate the currently active view since on mobile PWA experience address bar is not visible.

please review!

**Comment by stephenkingsley** [comment]
@addyosmani may you review again?

**Comment by addyosmani** [comment]
Thanks for updating, @stephenkingsley! I'm running into some interesting bugs around the routing when I try out a fresh profile or go to another browser and paste in:

https://hn-pwa-d8b2e.firebaseapp.com/newest/1
https://hn-pwa-d8b2e.firebaseapp.com/jobs/1
https://hn-pwa-d8b2e.firebaseapp.com/jobs/

image

**Comment by stephenkingsley** [comment]
@addyosmani I can not reappear it. I think maybe cache?

**Comment by addyosmani** [comment]
That is super strange. I guess it was a temporary deployment issue or as you said caching on my end. It appears to be fine now. I've verified https://github.com/tastejs/hacker-news-pwas/pull/73#issuecomment-320432891 look fine to me. 

The one final bug I'm seeing is to do with the loading spinner between views.

image

It looks like there's a scaling issue, but it only happens on desktop. The spinner scaling is absolutely fine on mobile and tablet. Could you take a look?

**Comment by stephenkingsley** [comment]
@addyosmani OK, I will modify the loading style.

**Comment by addyosmani** [comment]
Fixed on production now. Thanks, @stephenkingsley! Time to land this one 🚢

**Comment by addyosmani** [comment]
@stephenkingsley Just noticed a few more things:

image

The latest deploy is missing a 512px icon for the splashscreen. Can you get that added?

Can you also remove ` {"shrink-to-fit":"no"}.`?

**Comment by stephenkingsley** [comment]
yes! I will add the icon in today.

**Pull request merged**